naga/src/back/wgsl/writer.rs

```diff
@@ -481,15 +481,19 @@ impl<W: Write> Writer<W> {
         Ok(())
     }
 
-    fn write_type_inner(&mut self, module: &Module, inner: &TypeInner) -> BackendResult {
+    fn write_type_resolution(
+        &mut self,
+        module: &Module,
+        resolution: &proc::TypeResolution,
+    ) -> BackendResult {
         // This actually can't be factored out into a nice constructor method,
         // because the borrow checker needs to be able to see that the borrows
         // of `self.names` and `self.out` are disjoint.
         let type_context = WriterTypeContext {
             module,
             names: &self.names,
         };
-        type_context.write_type_inner(inner, &mut self.out)?;
```

Comment:
question(non-blocking): Would it be possible to retire `write_type_inner` entirely at some point?

Reply:
I started out with that as my goal. But one of the clienteles we need to serve is people doing debugging, and in some spots all you've got is a `TypeInner` and an arena.

Or at least I think so. Careful examination may show that, in every place we are currently dumping `TypeInner`, we actually have a `Handle<Type>` or a `TypeResolution` available as well. If that's the case, then we can totally make `write_type_inner` a private detail of `common::wgsl::types`.